include/zephyr/bluetooth/audio/tmap.h

```diff
@@ -0,0 +1,65 @@
+/** @file
+ *  @brief Header for Bluetooth TMAP.
+ *
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_TMAP_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_TMAP_
+
+#include <zephyr/sys/util.h>
+
+/** @brief TMAP Role characteristic */
+enum bt_tmap_role {
+	BT_TMAP_ROLE_CG = BIT(0),
+	BT_TMAP_ROLE_CT = BIT(1),
+	BT_TMAP_ROLE_UMS = BIT(2),
+	BT_TMAP_ROLE_UMR = BIT(3),
+	BT_TMAP_ROLE_BMS = BIT(4),
+	BT_TMAP_ROLE_BMR = BIT(5),
+};
+
+/** @brief TMAP callback structure. */
+struct bt_tmap_cb {
+	/** @brief TMAP discovery complete callback
+	 *
+	 *  This callback notifies the application about the value of the
+	 *  TMAP Role characteristic on the peer.
+	 *
+	 *  @param role		Peer TMAP role.
+	 *  @param conn     Pointer to the connection
+	 *  @param err      0 if success, -EINVAL if TMAS was not discovered.
```

Comment:
```suggestion
	 *  @param role     Peer TMAP role(s).
	 *  @param conn     Pointer to the connection
	 *  @param err      0 if success, -EINVAL if TMAS was not discovered.
```

Reply:
`EOPNOTSUPP` would also be better than `EINVAL` if we really want to return a errno value here. Alternatively we can just return the ATT error from the server